File: web/getpost.js
const PORT = 8000
const server = require('http') .createServer((req, res) => {
  res.setHeader('Set-Cookie', `last_access=${Date.now()};`);
  res.writeHead(200, {"Content-Type": "text/plain"})
  if (req.method == 'GET') {
    const body = ['get', req.headers.cookie, req.url.split('?').at(1)]
    res.end(body.join('\n'))
  }
  if (req.method == 'POST') {
    const body = ['post', req.headers.cookie]
    req.on('data', chunk => body.push(chunk))
    req.on('end', ()  => res.end(body.join('\n')))
  }
})
server.listen(PORT)

File: web/getpost.py
from http.server import BaseHTTPRequestHandler, HTTPServer
from urllib.parse import parse_qs, urlparse

class CustomHTTPRequestHandler(BaseHTTPRequestHandler):

    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type', 'text/plain; charset=utf-8')
        self.end_headers()
        body = f'GET:{urlparse(self.path).query}'
        self.wfile.write(body.encode())

    def do_POST(self):
        self.send_response(200)
        self.send_header('Content-Type', 'text/plain; charset=utf-8')
        self.end_headers()
        body = f'POST:{self.rfile.read(int(self.headers["content-length"])).decode("utf-8")}'
        self.wfile.write(body.encode())

PORT = 8000
SERVER = 'localhost'

server_address = (SERVER, PORT)
httpd = HTTPServer(server_address, CustomHTTPRequestHandler)
httpd.serve_forever()